tests/daemon/test.py

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import io
+import os
+from pathlib import Path
+from random import randint
+from subprocess import Popen
+from time import sleep
+from typing import Generator
+from urllib.error import URLError
+from urllib.request import urlopen
+
+from gql import Client, gql
+from gql.transport.requests import RequestsHTTPTransport
+from pytest import Config, fixture, mark
+
+
+def print_fd(fd: int):
+    os.lseek(fd, 0, io.SEEK_SET)
+    out_read = os.fdopen(fd, "r")
+    print(out_read.read())
+
+
+def check_server_up(port: int):
+    for _ in range(10):
+        try:
+            req = urlopen(f"http://127.0.0.1:{port}/", timeout=1.0)
+            if req.code == 200:
+                return
+            sleep(1.0)
+        except (URLError, TimeoutError):
+            sleep(1.0)
+    raise ValueError()
+
+
+@fixture
+def client(pytestconfig: Config, request) -> Generator[Client, None, None]:
+    port = randint(20000, 65000)
+    root = Path(pytestconfig.getoption("root"))
+    pipelines_dir = root / "share/revng/pipelines"
+    analyses_dir = root / "lib/revng/analyses"
+
+    env = {
+        **os.environ,
+        "PYTHONPATH": str((root / "lib/python").resolve()),
+        "REVNG_ANALYSIS_LIBRARIES": ":".join([str(p.resolve()) for p in analyses_dir.glob("*.so")]),
+        "REVNG_PIPELINES": ",".join([str(p.resolve()) for p in pipelines_dir.glob("*.yml")]),
+        "FLASK_RUN_PORT": str(port),
+        "FLASK_APP": "revng.daemon",
+        "FLASK_ENV": "development",
+    }
```

Comment:
Why do all of this when we can simply use `revng daemon`?

Reply:
That was before the command was implemented 😅 
Dropped